File: src/main/java/com/kaiosantiago/laudopro/entity/Customer.java
package com.kaiosantiago.laudopro.entity;

import com.kaiosantiago.laudopro.entity.base.BaseEntity;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
@Entity
@Table(name = "customer")
public class Customer extends BaseEntity {
    @Column(nullable = false, unique = true)
    private String fantasyName;

    @Column(nullable = false, unique = true)
    private String email;

    @Column(nullable = false, length = 14, unique = true)
    private String cnpj;

    @Column(nullable = false, length = 18, unique = true)
    private String cnpjFormated;

    @Column(nullable = false)
    private String address;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
}

File: LaudoPro.Domain/Models/Customer.cs
using LaudoPro.Domain.Models.Base;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.IO;
using System.Text.RegularExpressions;
using System.Xml.Linq;

namespace LaudoPro.Domain.Models
{
    [Table("customers")]
    public class Customer: BaseModel
    {
        [Required]
        public string FantasyName { get; set; }
        [Required]
        public string Email { get; set; }
        [Required]
        public string Cnpj{ get; set; }
        [Required]
        public string CnpjFormated { get; set; }
        [Required]
        public string Address { get; set; }
        [Required]
        public User User { get; set; }
    }
}
